src/util/exportToNDJson.js

```diff
@@ -114,6 +120,62 @@ const exportToNDJson = async (clientId, types, typeFilter, patient, systemLevelE
         }
       });
     }
+
+    const elementsQueries = {};
+    // create lookup object for _elements parameter
+    if (elements) {
+      elements.forEach(e => {
+        let elementNames = [];
+        if (e.includes('.')) {
+          const resourceType = e.split('.')[0];
+          const elementName = e.split('.')[1];
+          if (
+            Object.keys(choiceTypeElements[resourceType]).length !== 0 &&
+            Object.keys(choiceTypeElements[resourceType]).includes(`${elementName}[x]`)
+          ) {
+            const rootElem = elementName.split('[x]')[0];
+            choiceTypeElements[resourceType][rootElem].forEach(e => {
```

Comment:
This doesn't carefully check the choice type element structure exists before doing the `forEach`. 

It blows up on the following request
```
http://localhost:3000/$export?_type=DiagnosticReport&_elements=DiagnosticReport.effective
```

Additionally this area of the code could use a bit more explanation of what it is doing. If possible, unit tests too.

Reply:
Sounds good! I think I have a fix that I will push up. The issue was it wasn't finding the key with just the elementName in the lookup object (needed the [x] as well). I will add more documentation but will push that fix for now!

Reply:
I think it also needs to be a bit more resilient to garbage being passed in. Like a Resource or element that doesn't exist.

Reply:
How so? Should I check every resourceType and element that is passed in to make sure it's valid FHIR? If so, how does that work for elements passed in without resourceType (i.e. just `id` rather than `Conditiion.id`)? Would I have to check that it's valid on every single resourceType (often not the case)? These are things I have brought up but it seemed like the server should not error out.

Reply:
If you wanted to do a base check that something of the form `Condition.id` has id as a valid property on Condition, you could use the propertyPaths file/script here: https://github.com/projecttacoma/fqm-testify/commit/815c4fc5cef44f3f4056ce873f1b3678549fdb9b
Hasn't been pulled into fqm-testify because it's mostly just been used for the data requirements work so far.

Reply:
Do we have an idea of the expected error output according to the spec if there is malformed parameters?

Reply:
@hossenlopp @lmd59 also keep in mind that we don't do any sort of checks for inputs passed into `_typeFilter`. For example, a GET request to `$export?_typeFilter=Condition?invalid=invalid` does not return any errors.

Reply:
I think maybe some of this validation is out of scope for this tasking, but if we're not responding with the correct errors, we should know what the spec indicates we need and use that for potential follow-up tasking.

Reply:
Probably just 400 Bad Request, right?